src/qml/qmlwaveformrenderer.h

```diff
@@ -108,11 +124,95 @@ class QmlWaveformRendererRGB
     double m_gainMid;
     double m_gainHigh;
 
+    bool m_ignoreStem{false};
+
     ::WaveformRendererAbstract::PositionSource m_position{::WaveformRendererAbstract::Play};
     allshader::WaveformRendererSignalBase::Options m_options{
             allshader::WaveformRendererSignalBase::Option::None};
 };
 
+class QmlWaveformRendererRGB
+        : public QmlWaveformRendererSignal {
+    Q_OBJECT
+    QML_NAMED_ELEMENT(WaveformRendererRGB)
+
+  public:
+    Renderer create(WaveformWidgetRenderer* waveformWidget) const override;
+};
+
+class QmlWaveformRendererFiltered
+        : public QmlWaveformRendererSignal {
+    Q_OBJECT
+    Q_PROPERTY(bool stacked MEMBER m_stacked FINAL)
+
+    QML_NAMED_ELEMENT(WaveformRendererFiltered)
+
+  public:
+    Renderer create(WaveformWidgetRenderer* waveformWidget) const override;
+
+  private:
+    bool m_stacked{false};
+};
+
+class QmlWaveformRendererHSV
+        : public QmlWaveformRendererFactory {
+    Q_OBJECT
+    Q_PROPERTY(bool ignoreStem MEMBER m_ignoreStem NOTIFY ignoreStemChanged)
+    Q_PROPERTY(QColor axesColor MEMBER m_axesColor NOTIFY axesColorChanged REQUIRED)
+    Q_PROPERTY(QColor color MEMBER m_color NOTIFY colorChanged REQUIRED)
+    Q_PROPERTY(double gainAll MEMBER m_gainAll NOTIFY gainAllChanged REQUIRED)
+    Q_PROPERTY(double gainLow MEMBER m_gainLow NOTIFY gainLowChanged REQUIRED)
+    Q_PROPERTY(double gainMid MEMBER m_gainMid NOTIFY gainMidChanged REQUIRED)
+    Q_PROPERTY(double gainHigh MEMBER m_gainHigh NOTIFY gainHighChanged REQUIRED)
+    QML_NAMED_ELEMENT(WaveformRendererHSV)
+
+  public:
+    Renderer create(WaveformWidgetRenderer* waveformWidget) const override;
+  signals:
+    void axesColorChanged(const QColor&);
+    void colorChanged(const QColor&);
+    void ignoreStemChanged(bool);
+    void gainAllChanged(double);
+    void gainLowChanged(double);
+    void gainMidChanged(double);
+    void gainHighChanged(double);
+
+  private:
+    QColor m_axesColor;
+    QColor m_color;
+
+    double m_gainAll;
+    double m_gainLow;
+    double m_gainMid;
+    double m_gainHigh;
+
+    bool m_ignoreStem{false};
+};
+
+class QmlWaveformRendererSimple
+        : public QmlWaveformRendererFactory {
```

Comment:
whats teh difference between `QmlWaveformRendererFactory` and `QmlWaveformRendererSignal` again?

Wouldn't it be better to implement `ignoreStem` on `QmlWaveformRendererFactory`?

Reply:
`QmlWaveformRendererFactory` is the base class for any renderers, including pre/post rolls, beatgrid, markers, ...
`QmlWaveformRendererSignal` is the base class for any signal renderer (aka those that draw the waveform). `ignoreStem` only relevant for signal renderer (see `WaveformRendererSignalBase::setIgnoreStem`)

Reply:
Note that this is due to the fact that these renderers "skip" themselves if the loaded file is a stem, to let the stem renderer do the waveform drawing. Going forward, we could consider something driven from the top level instead (e.g some kind of conditional layer)